addons/sale_order_dates/views/sale_order_views.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0"?>
+<odoo>
+    <record id="view_order_form_inherit_sale_stock_inherit_sale_order_dates" model="ir.ui.view">
+        <field name="name">sale.order.form.inherit5</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale_stock.view_order_form_inherit_sale_stock"/>
+        <field name="arch" type="xml">
+            <group name="sale_shipping" position="inside">
+                <field name="requested_date" on_change="onchange_requested_date(requested_date, commitment_date)"/>
```

Comment:
forgot to remove on_change. it's now trigger by @api.onchange('requested_date') without defining in xml..